bndtools.m2e/src/bndtools/m2e/BndConfigurator.java

```diff
@@ -233,29 +252,66 @@ private void execJarMojo(final IMavenProjectFacade projectFacade, IProgressMonit
 		IMavenExecutionContext context = projectRegistryManager.createExecutionContext(projectFacade.getPom(),
 			resolverConfiguration);
 
+		List<File> outputDirectories = new ArrayList<>();
+
 		context.execute((context1, monitor1) -> {
+			IWorkspaceRoot wsroot = ResourcesPlugin.getWorkspace()
+				.getRoot();
 			SubMonitor progress = SubMonitor.convert(monitor1);
 			MavenProject mavenProject = getMavenProject(projectFacade, progress.newChild(1));
+			File buildDirectory = Central.getBuildDirectory(projectFacade.getProject());
 
 			List<MojoExecution> mojoExecutions = new ArrayList<>();
 			mojoExecutions.addAll(
 				projectFacade.getMojoExecutions("org.apache.maven.plugins", "maven-jar-plugin", monitor1, "jar"));
 			mojoExecutions.addAll(
 				projectFacade.getMojoExecutions("org.apache.maven.plugins", "maven-jar-plugin", monitor1, "test-jar"));
 
+			List<File> files = new ArrayList<>();
 			for (MojoExecution mojoExecution : mojoExecutions) {
 				maven.execute(mavenProject, mojoExecution, progress.newChild(1));
+
+				String goal = mojoExecution.getGoal();
+				File outputDirectory = maven.getMojoParameterValue( //
+					mavenProject, mojoExecution, //
+					((TEST_JAR.equals(goal)) ? TEST_CLASSES_DIRECTORY
+						: CLASSES_DIRECTORY), //
+					File.class, monitor);
+
+				outputDirectories.add(outputDirectory);
+
+				String artifactId = mavenProject.getArtifactId();
+				String finalName = ofNullable(
+					maven.getMojoParameterValue(mavenProject, mojoExecution, "finalName", String.class, monitor))
+						.orElse("");
+				String classifier = ofNullable(
+					maven.getMojoParameterValue(mavenProject, mojoExecution, "classifier", String.class, monitor))
+						.orElse("");
+
+				StringBuilder fileName = new StringBuilder(finalName);
+				if (!classifier.isEmpty()) {
+					fileName.append("-")
+						.append(classifier);
+					artifactId += "-" + classifier;
+				}
+				fileName.append(".jar");
+
+				files.add(new File(buildDirectory, fileName.toString()));
 			}
 
+			EclipseWorkspaceRepository workspaceRepo = Central.getEclipseWorkspaceRepository();
```

Comment:
I am unclear on the connection between EclipseWorkspaceRepository which is for Bnd Workspace projects and m2e which is for maven projects? EclipseWorkspaceRepository should assume that all projects in the map are Bnd workspace projects, no? Or are you trying to cross the streams here? (Ghostbusters reference).